domain/application/state/state.go

```diff
@@ -737,6 +737,11 @@ WHERE uuid = $charmID.uuid;
 		return nil, -1, fmt.Errorf("failed to get charm lxd profile: %w", err)
 	}
 
+	// Sometimes we get am empty slice, sometimes a nil slice.
+	// Cater for both cases.
+	if len(profile.LXDProfile) == 0 {
```

Comment:
This is annoying.

Reply:
Yeah, I added a TODO to figure out why

Reply:
This is because sqlite and dqlite differ. You can only run state tests with dqlite lite because they handle empty byte slices different. 

Remove this if statement.